### PR TITLE
feat[close #24] Add --dry-run option

### LINKED ISSUE #24: Add --dry-run option
This flag allow the transaction to be dropped at final steps. Useful for development purposes/tests.

### PR DESCRIPTION
**Pull request by mirkobrombin**
close #24

I am unable to test due a problem with ABRoot which afflict my installation since weeks:
```
  ERROR   building at STEP "LABEL maintainer 'Generated by ABRoot'": error decoding response from copier subprocess: invalid character '(' looking for beginning of value
Error: building at STEP "LABEL maintainer 'Generated by ABRoot'": error decoding response from copier subprocess: invalid character '(' looking for beginning of value
```

### CODE REVIEW

File: cmd/pkg.go [hunk 1]
```diff
@@ -33,6 +33,13 @@ func NewPkgCommand() *cmdr.Command {
 		pkg,
 	)
 
+	cmd.WithBoolFlag(
+		cmdr.NewBoolFlag(
+			"dry-run",
+			"d",
+			abroot.Trans("pkg.dryRunFlag"),
```
Comment (matbme) [inline cmd/pkg.go:40]: Needs to be added to `locales/en.yml`

File: cmd/upgrade.go [hunk 2]
```diff
@@ -40,6 +40,13 @@ func NewUpgradeCommand() *cmdr.Command {
 			abroot.Trans("upgrade.checkOnlyFlag"),
 			false))
 
+	cmd.WithBoolFlag(
+		cmdr.NewBoolFlag(
+			"dry-run",
+			"d",
+			abroot.Trans("upgrade.dryRunFlag"),
```
Comment (matbme) [inline cmd/upgrade.go:47]: Needs to be added to `locales/en.yml`

Comment (mirkobrombin) [reply cmd/upgrade.go:47]: added

File: cmd/pkg.go (hunk 1 above)
Comment (mirkobrombin) [reply cmd/pkg.go:40]: added

**Review by kbdharun — APPROVED** [review]
The locale changes LGTM, will lock the component on Weblate to prevent merge conficts after this PRs merge.

**Comment by matbme** [comment]
Needs some extra checks to avoid doing destructive operations:

- [This section](https://github.com/Vanilla-OS/ABRoot/blob/6137996fadc52acbf8cebfb8c14da34e396a9e37/core/system.go#L621-L625) deletes the current image, which should not happen in a dry-run
- [This entire section](https://github.com/Vanilla-OS/ABRoot/blob/6137996fadc52acbf8cebfb8c14da34e396a9e37/core/system.go#L690-L744) makes changes to the future partition
- [This part](https://github.com/Vanilla-OS/ABRoot/blob/6137996fadc52acbf8cebfb8c14da34e396a9e37/core/system.go#L750-L789) writes to the init partition
- [This part](https://github.com/Vanilla-OS/ABRoot/blob/6137996fadc52acbf8cebfb8c14da34e396a9e37/core/system.go#L791-L799) does more changes to the future partition

I think a dry-run should enter these functions and skip the write operation. If we modify part-future, the future partition may become unusable and it's not a dry-run anymore

**Review by matbme — APPROVED** [review]

**Pull request merged**